dataset.py

```diff
@@ -108,6 +108,33 @@ def random_subseq(sequence, seq_len):
     index = random.randint(0, len(sequence) - 1 - seq_len)
     return sequence[index:index + seq_len]
 
+def stretch_sequence(sequence, stretch_multiplier):
+    """ Iterate through sequence and stretch each time shift event by a factor """
+    time_sum = 0
+    time_count = 0
+    i = 0
+    while i < len(sequence):
+        evt = sequence[i]
+        if evt >= TIME_OFFSET and evt < VEL_OFFSET:
+            # Convert time event to number of seconds
+            time_sum += convert_time_evt_to_sec(evt)
+            time_count += 1
+            # TODO: Edge case where last event is time shift event
+        else:
+            if i > 0:
+                # Once there is a non time shift event, go backwards and recalculate previous sequential time shift events
+                for j in range(i - 1, i - 1 -time_count, -1):
```

Comment:
A more optimal algorithm will mark the first time a time event has been seen, so you don't need to do this loop. Instead, you just slice and replace a subsection of sequence. E.g:
```
sequence[first_evt_index:first_evt_index+time_count] = the replaced events
```